dask/utils.py

```diff
@@ -1032,3 +1035,29 @@ def __str__(self):
         return "<%s: %s>" % (self.__class__.__name__, self.token)
 
     __repr__ = __str__
+
+
+def effective_get(get=None, df=None):
+    """Get the effective get method used in a given situation"""
+    df_get = df._default_get if df else None
+    return get or _globals.get('get') or df_get
```

Comment:
I'm sort of +-0 on this generally.  I think that there is a cost to hiding this logic.  Inlining it is sometimes easier for other developers reading the code.  The benefit here is obviously consistency in the face of error prone logic.  The costs are readability and the need to traverse multiple files to tell what's going on.